packages/aws-cdk/lib/commands/flag-operations.ts

```diff
@@ -150,12 +143,30 @@ async function prototypeChanges(
     outdir: path.join(process.cwd(), 'original'),
   });
 
-  const cx = await toolkit.synth(source);
-  const assembly = cx.cloudAssembly;
-
   const updateObj: Record<string, boolean> = {};
-  updateObj[flagName] = boolValue;
+  if (typeof (flagNames) == 'string') {
```

Comment:
[nitpicking] I guess this works, but is a bit unusual. Why not make `flagNames` of type `string[]` and check if the length is 1?